### PR TITLE
refactor DefaultCaseDefinitionRepository and create generic method to…

### PR DESCRIPTION
**Pull request by aktaskaan**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CCD-4083

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -287,4 +288,7 @@ reference.data.cache.ttl.in.days=5
 # Refresh cache once everyday at 4am
 reference.data.cache.refresh.rate.cron=0 0 4 * * ?
 
+definition-store.retry.maxAttempts=${DEFINITION_STORE_RETRY_MAX_ATTEMPTS:2}
+definition-store.retry.maxDelay=${DEFINITION_STORE_RETRY_MAX_DELAY:1000}
+
```
Comment (RebeccaBaker) [inline src/main/resources/application.properties:294]: do we need to reference these HTTP_CLIENT_CONNECTION_DEFINITION_STORE_TIMEOUT, DEFINITION_STORE_RETRY_MAX_ATTEMPTS, DEFINITION_STORE_RETRY_MAX_DELAY in either the values.yaml or flux config?

Comment (aktaskaan) [reply src/main/resources/application.properties:294]: These are the default values and we will use flux when we need to change them.

Comment (aktaskaan) [reply src/main/resources/application.properties:294]: including HTTP_CLIENT_CONNECTION_DEFINITION_STORE_TIMEOUT

**Review by RebeccaBaker — APPROVED** [review]

**Review by dinesh1patel — APPROVED** [review]
LGTM

**Pull request merged**